meos/src/general/temporal.c

```diff
@@ -137,6 +125,51 @@ ensure_continuous_interpolation(int16 flags)
 }
 #endif /* not used */
 
+/**
+ * @brief Ensure that the interpolation is valid
+ * @note Used for the constructor functions
+ */
+void
+ensure_valid_interpolation(meosType temptype, interpType interp)
+{
+  if (interp == LINEAR && ! temptype_continuous(temptype))
+    ereport(ERROR, (errcode(ERRCODE_ARRAY_ELEMENT_ERROR),
+      errmsg("The temporal type cannot have linear interpolation")));
+  return;
+}
+
+/**
+ * @brief Ensure that the subtype of temporal type is a sequence (set)
+ */
+void
+ensure_continuous(const Temporal *temp)
+{
+  assert(temptype_subtype(temp->subtype));
+  if (temp->subtype == TINSTANT || MOBDB_FLAGS_GET_DISCRETE(temp->flags))
+    elog(ERROR, "Input must be a temporal continuous sequence (set)");
+  return;
+}
+
+/**
+ * @brief Ensure that two temporal values have the same interpolation
+ * @param[in] temp1,temp2 Input values
+ * @param[in] strict True when the interpolations must be equal, false when
+ * one of the input values may be of instant subtype and thus the interpolation
+ * would be INTERP_NONE which is compatible with any other interpolation
+ */
+void
+ensure_same_interpolation(const Temporal *temp1, const Temporal *temp2,
+  bool strict)
+{
+  if (! strict || (temp1->subtype == TINSTANT || temp2->subtype == TINSTANT))
+    return;
+  interpType interp1 = MOBDB_FLAGS_GET_INTERP(temp1->flags);
+  interpType interp2 = MOBDB_FLAGS_GET_INTERP(temp2->flags);
+  if (interp1 != interp2)
+    elog(ERROR, "The temporal values must have the same interpolation");
+  return;
+}
```

Comment:
What is the point of the `strict` parameter here?
If it is only used to tell if one of the parameters may be of instant subtype, we can probably compute it inside the function:
```c
bool strict = !(temp1->subtype == TINSTANT || temp2->subtype == TINSTANT);
```
In the current implementation, calling the function with `strict = false` does nothing.